distributed/worker.py

```diff
@@ -298,7 +298,11 @@ def _close(self, report=True, timeout=10):
             v.stop()
         self.rpc.close()
         self.status = 'closed'
-        self.stop()
+        with ignoring(gen.TimeoutError):
+            yield gen.with_timeout(timedelta(seconds=1),
+                    [close(stream) for stream in self._listen_streams])
+        for stream in self._listen_streams:
+            stream.close()
```

Comment:
@pitrou any thoughts on why this would cause more file descriptors to be left dangling?

The test `distributed/tests/test_client.py::test_open_close_many_workers[Nanny-10-20]` is now failing when we create and close many `Nanny`s.